GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java

```diff
@@ -196,10 +229,187 @@ private void writeSurvey(String title, File fileName,
+            HSSFCellStyle optionStyle, Map<Long, String> qTextFromId, QuestionDto q, HSSFRow row) {
+        createCell(row,  6, q.getTip(), null);
+        createCell(row,  7, q.getVariableName(), null);
+        createCell(row,  8, typeString(q), null);
+        createCell(row,  9, q.getMandatoryFlag(), null);
+        createCell(row, 10, q.getLocaleNameFlag(), null);
+        createCell(row, 11, q.getRequireDoubleEntry(), null);
+        // Dependency
+        createCell(row, 12, q.getDependentFlag(), null);
+        createCell(row, 13, qTextFromId.get(q.getDependentQuestionId()), null);
+        createCell(row, 14, q.getDependentQuestionAnswer(), null);
+        // Number
+        createCell(row, 15, q.getAllowSign(), null);
+        createCell(row, 16, q.getAllowDecimal(), null);
+        createCell(row, 17, q.getMinVal(), null);
+        createCell(row, 18, q.getMaxVal(), null);
+        // Option
+        createCell(row, 19, optionString(q), optionStyle); //TODO: wrap this cell?
+        createCell(row, 20, q.getType() == QuestionType.OPTION ?
+                q.getAllowMultipleFlag() : null, null);
+        createCell(row, 21, q.getAllowOtherFlag(), null);
+        // Geopos
+        createCell(row, 22, q.getLocaleLocationFlag(), null);
+        createCell(row, 23, q.getType() == QuestionType.GEO ? 
+                q.getGeoLocked() : null, null);
+        // CASCADE
+        createCell(row, 24, q.getCascadeResourceId(), null);                        
+        // geoshapes
+        createCell(row, 25, q.getAllowPoints(), null);
+        createCell(row, 26, q.getAllowLine(), null);
+        createCell(row, 27, q.getAllowPolygon(), null);
+        // caddisfly
+        createCell(row, 28, q.getCaddisflyResourceUuid(), null);
+        // barcode (just reusing other flags)
+        createCell(row, 29, q.getType() == QuestionType.SCAN ? 
+                q.getAllowMultipleFlag() : null, null);
+        createCell(row, 30, q.getType() == QuestionType.SCAN ? 
+                q.getGeoLocked() : null, null);
+    }
+
+    /**
+     * Creates the headers for the full-form export
+     * @param sheet
+     * @param startRow
+     * @param style
+     * @return
+     */
+    private int createFullHeader(HSSFSheet sheet, int startRow, String title, HSSFCellStyle style) {
+        int r = startRow;
+        int c = 0;
+        HSSFRow row = sheet.createRow(r++);
+        createCellBlock(row, c++, title, style, 6);
+
+        row = sheet.createRow(r++);
+        createCellBlock(row, 0, "Group", style, 3);
+        createCellBlock(row, 3, "Question", style, 3);
+        c = 12; //Group headers
+        createCellBlock(row, c, "Dependency", style, 3); // 3 wide
+        createCellBlock(row, c += 3, "Numbers", style, 4); // 4 wide
+        createCellBlock(row, c += 4, "Options", style, 3); // 3 wide
+        createCellBlock(row, c += 3, "Geolocation", style, 2); // 2 wide
+        createCell(     row, c += 2, "Cascade", style);
```

Comment:
isn't there too much spacing here after '('

Reply:
I wanted it to align vertically so you can see the pattern of it.